Objects/typeobject.c

```diff
@@ -5297,7 +5297,7 @@ get_base_by_token_recursive(PyTypeObject *type, void *token)
     Py_ssize_t n = PyTuple_GET_SIZE(bases);
     for (Py_ssize_t i = 0; i < n; i++) {
         PyTypeObject *base = _PyType_CAST(PyTuple_GET_ITEM(bases, i));
-        if (!_PyType_HasFeature(base, Py_TPFLAGS_HEAPTYPE)) {
+        if (!PyType_HasFeature(base, Py_TPFLAGS_HEAPTYPE)) {
```

Comment:
Why do you make this change? The underscore flavor is inlined and should be faster.

> Ensure the flag is checked by a function, not directly. PyType_Check()

The two functions are supposed to return the same result. It's not the case?

Reply:
Both are static inline functions, and now they have the same implementation with the same binary image.  It is not surprising for the internal version to switch back to a macro, which is a concern to me.  If the internal version can be faster, I would like an internal version of `PyType_Check()`.

That said, I'm fine with leaving a `_PyType_HasFeature()` in a loop.  I changed this for consistency in the related functions.

Reply:
The `telco` slowed down by 4% with `_PyType_HasFeature()`.  Maybe a noise, but I'll check later `PyType_GetModuleByDef()` with `PyType_HasFeature()`.

Reply:
`PyType_GetModuleByDef()` is slower by around 10% than expected.  It seems there are a few ways to recover it.

I'll make a new PR, taking back the change in `typeobject.c`.